addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -507,12 +507,18 @@ const Link = Widget.extend({
                 $link.html(this.data.originalHTML);
             } else if (linkInfos.content && linkInfos.content.length) {
                 let contentWrapperEl = $link[0];
-                // Update the first child element that has the same inner text
+                const text = $link[0].innerText.replaceAll("\u200B", "").trim();
+                let child = [...contentWrapperEl.children].find(
+                    (child) => !child.hasAttribute("data-o-link-zws")
+                );
+                // Update the first not ZWS child element that has the same inner text
                 // as the link with the new content while preserving child
                 // elements within the link. (e.g. the link is bold and italic)
-                while (contentWrapperEl.firstElementChild
-                    && (contentWrapperEl.firstElementChild.innerText === $link[0].innerText)) {
-                    contentWrapperEl = contentWrapperEl.firstElementChild;
+                while (child?.innerText.replaceAll("\u200B", "").trim() === text) {
+                    contentWrapperEl = child;
+                    child = [...contentWrapperEl.children].find(
+                        (child) => !child.hasAttribute("data-o-link-zws")
+                    );
```

Comment:
If I'm not mistaken you could get it to have that `find` function only once if you used a `do...while` instead:

```suggestion
                // Update the first not ZWS child element that has the same inner text
                // as the link with the new content while preserving child
                // elements within the link. (e.g. the link is bold and italic)
                let child;
                do {
                    contentWrapperEl = child || contentWrapperEl;
                    child = [...contentWrapperEl.children].find(child => !child.hasAttribute('data-o-link-zws'));
                } while (child?.innerText.replaceAll('\u200B', '').trim() === text);
```

@dmo-odoo Would you consider that golfing or do you agree it's better that way? :eyes:

Reply:
I'm confused here. How is this equivalent? `child` is not defined in your `while` condition.

Also, I don't actually want to golf, I just wanted to avoid having that `find` condition twice but I was worried you'd think my suggestion was too "golfy" :-P

Reply:
So should I golf it or not  here

Reply:
Yeah sorry, I meant `node`, obviously.
```js
let node = contentWrapperEl;
do {
    node = node.querySelector(':scope > :not([data-o-link-zws])');
} while (node?.innerText.replaceAll('\u200B', '').trim() === text);
```
In fact, the replace+trim+text is not really needed either if I understand the issue correctly:
```js
let node = contentWrapperEl;
do {
    node = node.querySelector(':scope > :not([data-o-link-zws])');
} while (node?.innerText.includes('\u200B');
```

Reply:
Let's discuss about it all together on Thursday then.

Reply:
@Mtaylorr We talked with @Zynton and I understand the issue better now. Let's go with the original suggestion from @Zynton at https://github.com/odoo/odoo/pull/155038#discussion_r1512850946.

Note that, as opposed to my suggestions on the matter, the change suggested by @Zynton was not merely for the sake of style. If someone ever needs to change anything in the callback of the `find` function, they might not realize, or forget, that they also have to make the change in the other one. This is especially true if they do the fix a tad too quickly, or if the reported issue only pertains to one of the two `find` while in fact the same bug might happen with the other `find` as well and is simply still waiting to be discovered.

Reply:
Just a tiny change, now I read it again, best to change the name of the argument in the `find` because otherwise we end up with to different variables called `child` and it might get confusing.

```suggestion
                // Update the first not ZWS child element that has the same inner text
                // as the link with the new content while preserving child
                // elements within the link. (e.g. the link is bold and italic)
                let child;
                do {
                    contentWrapperEl = child || contentWrapperEl;
                    child = [...contentWrapperEl.children].find(
                        element => !element.hasAttribute('data-o-link-zws'))
                    );
                } while (child?.innerText.replaceAll('\u200B', '').trim() === text);
```